test/com/linkedin/drelephant/schedulers/OozieSchedulerTest.java

```diff
@@ -0,0 +1,182 @@
+package com.linkedin.drelephant.schedulers;
+
+import com.linkedin.drelephant.configurations.scheduler.SchedulerConfigurationData;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import mockit.Expectations;
+import mockit.Mocked;
+import org.junit.Test;
+import org.apache.oozie.client.OozieClient;
+import org.apache.oozie.client.OozieClientException;
+import org.apache.oozie.client.WorkflowJob;
+
+import static org.junit.Assert.assertEquals;
+
+
+public class OozieSchedulerTest {
+  @Mocked
+  OozieClient oozieClient;
+
+  @Mocked
+  WorkflowJob workflowJob;
+
+  @Test
+  public void testOozieLoadInfoWithCompleteConf() throws Exception {
+    new Expectations() {{
+      workflowJob.getAppName();
+      result = "some-workflow-name";
+
+      oozieClient.getJobInfo("0004167-160629080632562-oozie-oozi-W");
+      result = workflowJob;
+    }};
+
+    SchedulerConfigurationData schedulerConfig = makeConfig("oozie", getDefaultSchedulerParams());
+    OozieScheduler scheduler = new OozieScheduler("id", getOozieProperties(), schedulerConfig);
+
+    assertEquals("http://oozie/search?workflow=some-workflow-name", scheduler.getFlowDefUrl());
+    assertEquals("some-workflow-name", scheduler.getFlowDefId());
+    assertEquals("http://oozie/workflows/0004167-160629080632562-oozie-oozi-W", scheduler.getFlowExecUrl());
+    assertEquals("0004167-160629080632562-oozie-oozi-W", scheduler.getFlowExecId());
+
+    assertEquals("http://oozie/search?workflow=some-action", scheduler.getJobDefUrl());
+    assertEquals("some-action", scheduler.getJobDefId());
+    assertEquals("http://oozie/workflows/0004167-160629080632562-oozie-oozi-W@some-action", scheduler.getJobExecUrl());
+    assertEquals("0004167-160629080632562-oozie-oozi-W@some-action", scheduler.getJobExecId());
+
+    assertEquals("some-action", scheduler.getJobName());
+    assertEquals(0, scheduler.getWorkflowDepth());
+    assertEquals("oozie", scheduler.getSchedulerName());
+  }
+
+  @Test(expected = RuntimeException.class)
```

Comment:
I would use an `ExpectedException` `@Rule` rather than this notation. IMHO this way of doing things is a little heavy handed because if the exception occurs anywhere within the method the test will pass. This is especially problematic with something as generic as a `RuntimeExecpetion`. If you use a rule you can be confident about where the exception occurred. Also, you can specify the expected message to be even more confident that you are getting the exception that you think you are getting. See `FetcherConfigurationTest.java` for an example.